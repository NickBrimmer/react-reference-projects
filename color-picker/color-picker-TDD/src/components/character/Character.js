import React, { Component } from 'react';
import PropTypes from 'prop-types';

// function -> class
// extends Component
// render method returns jsx
// move propTypes into static property

export default class Character extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    myFunc: PropTypes.func.isRequired
  };

  render() {
    const { name, species, status, image, myFunc } = this.props;

    return (
      <dl>
        <dt onClick={() => myFunc(name)}>Name</dt>
        <dd>{name}</dd>

        <dt>Species</dt>
        <dd>{species}</dd>

        <dt>Status</dt>
        <dd>{status}</dd>

        <dt>Image</dt>
        <dd><img src={image} /></dd>
      </dl>
    );
  }
}

// const Character = ({ name, species, status, image, myFunc }) => {
//   return (
//     <dl>
//       <dt onClick={() => myFunc(name)}>Name</dt>
//       <dd>{name}</dd>

//       <dt>Species</dt>
//       <dd>{species}</dd>

//       <dt>Status</dt>
//       <dd>{status}</dd>

//       <dt>Image</dt>
//       <dd><img src={image} /></dd>
//     </dl>
//   );
// };
