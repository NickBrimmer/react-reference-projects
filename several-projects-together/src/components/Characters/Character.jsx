import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, species }) => (
  <>  
    <h3>{name}</h3>
    <p>{species}</p>
  </>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired
};

export default Character;
