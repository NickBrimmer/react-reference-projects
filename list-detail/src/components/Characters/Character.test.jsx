import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character image="my image" name="Cool" description="hello" />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
