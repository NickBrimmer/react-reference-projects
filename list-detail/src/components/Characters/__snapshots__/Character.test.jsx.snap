// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<section
  className="Character"
>
  <img
    alt="Cool"
    src="my image"
  />
  <h3>
    Cool
  </h3>
  <p>
    hello
  </p>
</section>
`;
